libraries/rib-base/src/main/java/com/badoo/ribs/core/Node.kt

```diff
@@ -345,6 +346,23 @@ open class Node<V : RibView> @VisibleForTesting internal constructor(
             || subtreeHandlers.any { it.handleBackPressFallback() }
     }
 
+    fun upNavigation() {
+        when {
+            isRoot -> integrationPoint.handleUpNavigation()
+
+            else -> parent?.handleUpNavigation()
+                ?: error("If not root, Node should have a parent")
```

Comment:
Use `handleFatalError`?

Reply:
Replace warning message with `Can't handle up navigation, Node is not a root and has no parent`.